manager/integration/tests/test_infra.py

```diff
@@ -0,0 +1,129 @@
+import os
+import pytest
+import time
+
+from digitalocean import digitalocean
+from common import get_core_api_client, get_longhorn_api_client
+from common import RETRY_COUNTS, RETRY_INTERVAL
+
+
+def detect_cloudprovider():
+    cloudprovider_name = os.getenv("CLOUDPROVIDER")
+
+    if cloudprovider_name == "digitalocean":
+        cloudprovider = digitalocean()
+        return cloudprovider
+
+
+def is_node_ready_k8s(node_name):
+    k8s_api_client = get_core_api_client()
+
+    node_status_k8s = k8s_api_client.read_node_status(node_name)
+
+    for node_condition in node_status_k8s.status.conditions:
+        if node_condition.type == "Ready" and \
+           node_condition.status == "True":
+            node_ready = True
+        else:
+            node_ready = False
+
+    return node_ready
+
+
+def is_node_ready_longhorn(node_name):
+    longhorn_api_client = get_longhorn_api_client()
+
+    for longhorn_node in longhorn_api_client.list_node():
```

Comment:
We should able to get the node by name? Shouldn't need a list.

Also, if the node is not found, we should error out.

Reply:
Unresolved.

Reply:
done